activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -378,6 +378,10 @@ def adapter_name
         self.class::ADAPTER_NAME
       end
 
+      def connected?
+        true
```

Comment:
AbstractAdapter owns `@raw_connection`, so I think this can do the

```suggestion
        !@raw_connection.nil?
```